### PR TITLE
 JAMES-2871 Fix delimiter while loading env list variable

### PR DESCRIPTION
**Pull request by trantienduchn**

### CODE REVIEW

File: server/container/guice/configuration/src/main/java/org/apache/james/utils/PropertiesWrapperConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+
+import org.apache.commons.configuration2.Configuration;
+import org.apache.commons.configuration2.ConfigurationDecoder;
+import org.apache.commons.configuration2.ImmutableConfiguration;
+import org.apache.commons.configuration2.interpol.ConfigurationInterpolator;
+import org.apache.commons.configuration2.interpol.Lookup;
+import org.apache.commons.configuration2.sync.LockMode;
+import org.apache.commons.configuration2.sync.Synchronizer;
+
+public class PropertiesWrapperConfiguration implements Configuration {
```
Comment (chibenwa) [inline server/container/guice/configuration/src/main/java/org/apache/james/utils/PropertiesWrapperConfiguration.java]: This looks like a "delegate"

**Comment by chibenwa** [comment]
```
[7d8b86b6a182299c6505451769024ddf008b5a6e] + docker logs james-server-cassandra-rabbitmq-fix-delimiter-77b523ec-28c9-491f-adcb-2320720f5ee8
[7d8b86b6a182299c6505451769024ddf008b5a6e] SLF4J: Class path contains multiple SLF4J bindings.
[7d8b86b6a182299c6505451769024ddf008b5a6e] SLF4J: Found binding in [jar:file:/root/james-server-cassandra-rabbitmq-guice.lib/slf4j-log4j12-1.7.27.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[7d8b86b6a182299c6505451769024ddf008b5a6e] SLF4J: Found binding in [jar:file:/root/james-server-cassandra-rabbitmq-guice.lib/logback-classic-1.2.3.jar!/org/slf4j/impl/StaticLoggerBinder.class]
[7d8b86b6a182299c6505451769024ddf008b5a6e] SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
[7d8b86b6a182299c6505451769024ddf008b5a6e] SLF4J: Actual binding is of type [org.slf4j.impl.Log4jLoggerFactory]
[7d8b86b6a182299c6505451769024ddf008b5a6e] log4j:WARN No appenders could be found for logger (org.apache.commons.beanutils.converters.BooleanConverter).
[7d8b86b6a182299c6505451769024ddf008b5a6e] log4j:WARN Please initialize the log4j system properly.
[7d8b86b6a182299c6505451769024ddf008b5a6e] log4j:WARN See http://logging.apache.org/log4j/1.2/faq.html#noconfig for more info.
[7d8b86b6a182299c6505451769024ddf008b5a6e] Exception in thread "main" com.google.inject.ProvisionException: Unable to provision, see the following errors:
[7d8b86b6a182299c6505451769024ddf008b5a6e] 
[7d8b86b6a182299c6505451769024ddf008b5a6e] 1) Error in custom provider, java.lang.NullPointerException
[7d8b86b6a182299c6505451769024ddf008b5a6e]   at org.apache.james.modules.mailbox.ElasticSearchMailboxModule.getElasticSearchConfiguration(ElasticSearchMailboxModule.java:60) (via modules: com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.modules.mailbox.ElasticSearchMailboxModule)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.backends.es.ElasticSearchConfiguration
[7d8b86b6a182299c6505451769024ddf008b5a6e]     for the 1st parameter of org.apache.james.backends.es.ClientProvider.<init>(ClientProvider.java:50)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   at org.apache.james.modules.mailbox.ElasticSearchClientModule.configure(ElasticSearchClientModule.java:32) (via modules: com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$OverrideModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> com.google.inject.util.Modules$CombinedModule -> org.apache.james.modules.mailbox.ElasticSearchClientModule)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.backends.es.ClientProvider
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.elasticsearch.client.RestHighLevelClient
[7d8b86b6a182299c6505451769024ddf008b5a6e]     for the 1st parameter of org.apache.james.modules.mailbox.ElasticSearchStartUpCheck.<init>(ElasticSearchStartUpCheck.java:35)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.modules.mailbox.ElasticSearchStartUpCheck
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.lifecycle.api.StartUpCheck annotated with @com.google.inject.internal.Element(setName=,uniqueId=58, type=MULTIBINDER, keyType=)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating java.util.Set<org.apache.james.lifecycle.api.StartUpCheck>
[7d8b86b6a182299c6505451769024ddf008b5a6e]     for the 1st parameter of org.apache.james.StartUpChecksPerformer$StartUpChecks.<init>(StartUpChecksPerformer.java:81)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.StartUpChecksPerformer$StartUpChecks
[7d8b86b6a182299c6505451769024ddf008b5a6e]     for the 1st parameter of org.apache.james.StartUpChecksPerformer.<init>(StartUpChecksPerformer.java:117)
[7d8b86b6a182299c6505451769024ddf008b5a6e]   while locating org.apache.james.StartUpChecksPerformer
[7d8b86b6a182299c6505451769024ddf008b5a6e] 
[7d8b86b6a182299c6505451769024ddf008b5a6e] 1 error
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at com.google.inject.internal.InternalProvisionException.toProvisionException(InternalProvisionException.java:226)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at com.google.inject.internal.InjectorImpl$1.get(InjectorImpl.java:1053)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at com.google.inject.internal.InjectorImpl.getInstance(InjectorImpl.java:1086)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.GuiceJamesServer.start(GuiceJamesServer.java:81)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.CassandraRabbitMQJamesServerMain.main(CassandraRabbitMQJamesServerMain.java:46)
[7d8b86b6a182299c6505451769024ddf008b5a6e] Caused by: java.lang.NullPointerException
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.utils.PropertiesWrapperConfiguration.splitAndStripDoubleQuotes(PropertiesWrapperConfiguration.java:374)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.utils.PropertiesWrapperConfiguration.getStringArray(PropertiesWrapperConfiguration.java:277)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.backends.es.ElasticSearchConfiguration.getHosts(ElasticSearchConfiguration.java:149)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.backends.es.ElasticSearchConfiguration.fromProperties(ElasticSearchConfiguration.java:136)
[7d8b86b6a182299c6505451769024ddf008b5a6e] 	at org.apache.james.modules.mailbox.ElasticSearchMailboxModule.getElasticSearchConfiguration(ElasticSearchMailboxModule.java:161)
```

Please @trantienduchn fix this issue

**Comment by mbaechler** [comment]
The list splitting is done in AbstractConfiguration::addPropertyInternal , before the env variable is resolved by EnvironmentVariableStringLookup so there's no way to configure it properly.
However, repeating a key in the property file make it possible to define a list. We could ask ops team if it's doable that way.

**Comment by mbaechler** [comment]
> The list splitting is done in AbstractConfiguration::addPropertyInternal , before the env variable is resolved by EnvironmentVariableStringLookup so there's no way to configure it properly.
However, repeating a key in the property file make it possible to define a list. We could ask ops team if it's doable that way.

They prefer the current solution so we'll have to merge this PR.

**Comment by chibenwa** [comment]
`MemoryUploadCucumberTest » Socket Broken pipe (Write failed)` 

test this please

**Comment by trantienduchn** [comment]
```
[3b0bd0a7847146f3b05f6090e76cf8d971fee6b6] [ERROR] Errors: 
[3b0bd0a7847146f3b05f6090e76cf8d971fee6b6] [ERROR]   MemoryUploadCucumberTest » Socket Broken pipe (Write failed)
```
test this please

**Comment by trantienduchn** [comment]
```
[3b0bd0a7847146f3b05f6090e76cf8d971fee6b6] [ERROR] Errors: 
[3b0bd0a7847146f3b05f6090e76cf8d971fee6b6] [ERROR]   MemoryUploadCucumberTest » Socket Broken pipe (Write failed)
```
test this please

**Pull request closed without merging**